canary/archival.go

```diff
@@ -158,6 +190,85 @@ func archivalExternalWorkflow(ctx workflow.Context, scheduledTimeNanos int64) er
 	return profile.end(nil)
 }
 
-func archivalLargeResultActivity() ([]byte, error) {
+func largeResultActivity() ([]byte, error) {
 	return make([]byte, resultSize, resultSize), nil
 }
+
+func visibilityArchivalActivity(ctx context.Context, scheduledTimeNanos int64) error {
+	scope := activity.GetMetricsScope(ctx)
+	var err error
+	scope, sw := recordActivityStart(scope, activityTypeVisibilityArchival, scheduledTimeNanos)
+	defer recordActivityEnd(scope, sw, err)
+
+	client := getActivityArchivalContext(ctx).cadence
+	resp, err := client.Describe(ctx, archivalDomain)
+	if err != nil {
+		return err
+	}
+
+	if resp.Configuration != nil &&
+		resp.Configuration.GetVisibilityArchivalStatus() == shared.ArchivalStatusDisabled {
+		return errors.New("domain not configured for visibility archival")
+	}
+
+	visArchivalURI := ""
+	if resp.Configuration != nil {
+		visArchivalURI = resp.Configuration.GetVisibilityArchivalURI()
+	}
+
+	scheme := getURIScheme(visArchivalURI)
```

Comment:
Users don't actually have do this right? You aren't using scheme anywhere in the request? 

I think unless we are really testing something by verifying this scheme we should just have canary interact with this feature like a user would.

Reply:
User won't do this as we ensure the right URI is set for them. But for canary, I think it makes sense to perform this check so can we can identify the problem faster when the test fails (either due to misconfiguration or there's something wrong with visibility archival).